src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -999,6 +999,15 @@ void shim::xclSysfsGetDeviceInfo(xclDeviceInfo2 *info)
     info->mNumClocks = numClocks(info->mName);
 
     mDev->sysfs_get<unsigned short>("mb_scheduler", "kds_numcdmas", errmsg, info->mNumCDMA, static_cast<unsigned short>(-1));
+    
+    mDev->sysfs_get("", "link_width", errmsg, info->mPCIeLinkWidth, static_cast<unsigned short>(-1));
+    mDev->sysfs_get("", "link_speed", errmsg, info->mPCIeLinkSpeed, static_cast<unsigned short>(-1));
+    mDev->sysfs_get("", "link_speed_max", errmsg, info->mPCIeLinkSpeedMax, static_cast<unsigned short>(-1));
+    mDev->sysfs_get("", "link_width_max", errmsg, info->mPCIeLinkWidthMax, static_cast<unsigned short>(-1));
+    
+    //dont try to get any information which needs mailbox communication when device is not ready.
+    if(!mDev->is_mgmt() && !mDev->is_ready)
```

Comment:
From a flow perspective, this code looks good.   

From a initialization perspective, there are variables in the info structure that are NOT initialized, which will result in unexpected behavior.  It would be better if these values were initialized.

Normally, it would be great to initialize all of these values at the beginning of this method, but looking at the flow suggests doing this blindly will not work for methods calling this method does set some values.

Being that we are nearly the end-of-the-release, it would be better to do the following:
1.  Move line 1009 and invert the logic to 1017.  The idea here is to read the sensor only when the board is in a good state.  
2. Remove the return on line 1010.
2. Initialize the variables on line 1013 to zero.  Simply adding an `= -1` should work, assuming that indeed that is what the default variable should be.

Doing this will allow for more consistent initialization of this structure.

Reply:
Thank you Vamshi for review the code with me and showing:

1. The device structure is initialized to zeros prior to calling these function.
2. The `sysfs_get_sensor()` function returns back a zero if errors occur.
3. The reports are never called if the device is not ready.

For these reasons, the issues I had earlier have been resolved.